main/OpenCover.Framework/Manager/ProfilerManager.cs

```diff
@@ -311,11 +319,10 @@ private static void ConsumeException(Action doSomething)
                             }
                             break;
                         case 2:
-                            threadTerminated.Set();
+                            threadTermination.ThreadFinishedEvent.Set();
                             return;
                     }
                 }
-                threadTerminated.Set();
                 _memoryManager.RemoveDeactivatedBlocks();
```

Comment:
but something could be added in future.
if I read code and see ```ThreadFinishedEvent``` I expect that it would be set in all paths of thread execution, not only one path, that is checked by existing code. Exception occurred path also doesn't set this event BTW.